python/paddle/fluid/incubate/fleet/parameter_server/distribute_transpiler/distributed_strategy_factory.py

```diff
@@ -0,0 +1,221 @@
+                else:
+                    raise ValueError(
+                        "ServerRuntimeConfig doesn't have key: {}".format(key))
+        else:
+            raise TypeError(
+                "server_runtime_config only accept input type: dict or ServerRuntimeConfig"
+            )
+
+    def get_execute_strategy(self):
+        return self._execute_strategy
+
+    def set_execute_strategy(self, config):
+        if isinstance(config, fluid.ExecutionStrategy):
+            self._execute_strategy = config
+        elif isinstance(config, dict):
+            for key in config:
+                if hasattr(self._execute_strategy, key):
+                    setattr(self._execute_strategy, key, config[key])
+                else:
+                    raise ValueError(
+                        "ExecutionStrategy doesn't have key: {}".format(key))
+        else:
+            raise TypeError(
+                "execute_strategy only accept input type: dict or ExecutionStrategy"
+            )
+
+    def get_build_strategy(self):
+        return self._build_strategy
+
+    def set_build_strategy(self, config):
+        if isinstance(config, fluid.BuildStrategy):
+            self._build_strategy = config
+        elif isinstance(config, dict):
+            for key in config:
+                if hasattr(self._build_strategy, key):
+                    setattr(self._build_strategy, key, config[key])
+                else:
+                    raise ValueError(
+                        "BuildStrategy doesn't have key: {}".format(key))
+        else:
+            raise TypeError(
+                "build_strategy only accept input type: dict or BuildStrategy")
+
+
+class SyncStrategy(DistributedStrategy):
+    def __init__(self):
+        super(SyncStrategy, self).__init__()
+        self._program_config.sync_mode = True
+        self._program_config.runtime_split_send_recv = False
+        self._build_strategy.async_mode = False
+
+
+class AsyncStrategy(DistributedStrategy):
+    def __init__(self):
+        super(AsyncStrategy, self).__init__()
+        self._program_config.sync_mode = False
+        self._program_config.runtime_split_send_recv = True
+        self._build_strategy.async_mode = True
+
+
+class HalfAsyncStrategy(DistributedStrategy):
+    def __init__(self):
+        super(HalfAsyncStrategy, self).__init__()
+        self._program_config.sync_mode = False
+        self._program_config.runtime_split_send_recv = False
+        self._build_strategy.async_mode = False
+
+
+class GeoStrategy(DistributedStrategy):
+    def __init__(self, update_frequency=100):
+        super(GeoStrategy, self).__init__()
+        self._program_config.sync_mode = False
+        self._program_config.runtime_split_send_recv = True
+        self._program_config.geo_sgd_mode = True
+        self._program_config.geo_sgd_need_push_nums = update_frequency
+        self._build_strategy.async_mode = True
+
+
+class DistributedStrategyFactory(object):
```

Comment:
rename `DistributedStrategyFactory` to `DistributedStrategy` and rename `DistributedStrategy` to another name.

Reply:
DistributedStrategyFactory is renamed to StrategyFactory